--- a/lib/python/Components/Harddisk.py	2012-10-04 16:43:17.000000000 +0200
+++ b/lib/python/Components/Harddisk.py	2012-10-04 17:18:03.000000000 +0200
@@ -6,6 +6,7 @@
 from SystemInfo import SystemInfo
 from Components.Console import Console
 import Task
+from Tools.HardwareInfo import HardwareInfo
 
 def readFile(filename):
 	file = open(filename)
@@ -122,7 +123,11 @@
 		elif self.type == DEVTYPE_DEVFS:
 			ide_cf = self.device[:2] == "hd" and "host0" not in self.dev_path
 
-		internal = "pci" in self.phys_path
+		hw_type = HardwareInfo().get_device_name()
+		if hw_type == 'elite' or hw_type == 'premium' or hw_type == 'premium+' or hw_type == 'ultra' :
+			internal = "ide" in self.phys_path
+		else:
+			internal = "pci" in self.phys_path
 
 		if ide_cf:
 			ret = "External (CF)"
@@ -673,6 +678,9 @@
 				physdev = dev
 				print "couldn't determine blockdev physdev for device", device
 		error, blacklisted, removable, is_cdrom, partitions, medium_found = self.getBlockDevInfo(device)
+		hw_type = HardwareInfo().get_device_name()
+		if hw_type == 'elite' or hw_type == 'premium' or hw_type == 'premium+' or hw_type == 'ultra' :
+			if device[0:3] == "hda": blacklisted = True
 		if not blacklisted and medium_found:
 			description = self.getUserfriendlyDeviceName(device, physdev)
 			p = Partition(mountpoint = self.getMountpoint(device), description = description, force_mounted = True, device = device)

